From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: 404Setup <153366651+404Setup@users.noreply.github.com>
Date: Sun, 7 Jul 2024 20:40:23 +0800
Subject: [PATCH] Mobtimizations: RePathing optimization

Original code by Corosauce
You can find the original code on https://github.com/Corosauce/mobtimizations

disables the frequent pathfinds done while already following a path

diff --git a/src/main/java/net/minecraft/world/entity/ai/navigation/PathNavigation.java b/src/main/java/net/minecraft/world/entity/ai/navigation/PathNavigation.java
index 544920a31b649985333f82beafa94a3392f5853e..f014649faee00cbacf6a92cf327d71a7c53c1ac4 100644
--- a/src/main/java/net/minecraft/world/entity/ai/navigation/PathNavigation.java
+++ b/src/main/java/net/minecraft/world/entity/ai/navigation/PathNavigation.java
@@ -80,6 +80,7 @@ public abstract class PathNavigation {
     }
 
     public void recomputePath() {
+        if (one.tranic.vine.config.VineConfig.Optimize.mobRePathfinding) return; // Vine
         if (this.level.getGameTime() - this.timeLastRecompute > 20L) {
             if (this.targetPos != null) {
                 this.path = null;
@@ -437,6 +438,7 @@ public abstract class PathNavigation {
     }
 
     public boolean shouldRecomputePath(BlockPos pos) {
+        if (one.tranic.vine.config.VineConfig.Optimize.mobRePathfinding) return false; // Vine
         if (this.hasDelayedRecomputation) {
             return false;
         } else if (this.path != null && !this.path.isDone() && this.path.getNodeCount() != 0) {
